core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -998,7 +998,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    */
   @DeveloperApi
   override def requestExecutors(numAdditionalExecutors: Int): Boolean = {
-    assert(master.contains("yarn") || dynamicAllocationTesting,
+    assert(master.contains("mesos") || master.contains("yarn") || dynamicAllocationTesting,
       "Requesting executors is currently only supported in YARN mode")
```

Comment:
Change this message to be "... only supported in YARN or Mesos modes", and the message below